packages/marshal/src/rankOrder.js

```diff
@@ -0,0 +1,300 @@
+// @ts-check
+
+import { assert, details as X, q } from '@agoric/assert';
+import { getTag } from './helpers/passStyle-helpers.js';
+import { nameForPassableSymbol } from './helpers/symbol.js';
+import { Far } from './make-far.js';
+import { makeTagged } from './makeTagged.js';
+import { passStyleOf } from './passStyleOf.js';
+import { sameValueZero } from './pureCopy.js';
+
+const {
+  getOwnPropertyDescriptors,
+  fromEntries,
+  entries,
+  setPrototypeOf,
+} = Object;
+
+const { ownKeys } = Reflect;
+
+const firstSymbol = Symbol.for('');
+const firstTagged = makeTagged('', null);
+const aFar = Far('after taggeds', {});
+const anError = new Error('error sigil');
+const aPromise = Promise.resolve('promise sigil');
+
+/**
+ * Aside from 'undefined', lists the PassStyles from `types.js` in the same
+ * order as they are declared there. This list and that declaration must be
+ * co-maintained.
+ *
+ * JavaScript `Array.prototype.sort` sorts all `undefined`s to the end of the
+ * array without consulting the compare function. To avoid fighting with it,
+ * we put `undefined` at the end, where it can serve as the end-sigil of the
+ * ordering. In the same sense, `null` is the start-sigil of the ordering. This
+ * is useful for range searches.
+ *
+ * @type {[PassStyle, RankCover][]}
```

Comment:
This type should be enough to ensure the declaration and implementation are properly in sync.

Reply:
I tested a bit. I made some mods to `PassStyleRankAndCover` contents in my IDE to see what would happen. It complained each time as I had hoped. I think this is working. Resolving.